packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -287,559 +295,688 @@ void main() {
+            buildInputDecorator(
+              decoration: const InputDecoration(
+                filled: true,
+                labelText: labelText,
+                helperText: helperText,
+              ),
+            ),
+          );
 
-      expect(getDecoratorRect(tester).size, const Size(800.0, 56.0));
-      // Label line height is forced to 1.0 and font size is 16.0,
-      // the label should be vertically centered (20 pixels above and below).
-      expect(getLabelRect(tester).top, 20.0);
-      expect(getLabelRect(tester).bottom, 36.0);
-      // From the M3 specification, centering the label is right, but setting the line height to 1.0 is not
-      // compliant (the expected text style is bodyLarge which font size is 16.0 and its line height 1.5).
-      // TODO(bleroux): fix input decorator to not rely on forcing the label text line height to 1.0.
-    });
+          expect(getContainerRect(tester).height, containerHeight);
+          expect(getBorder(tester), isA<UnderlineInputBorder>());
+          expect(getBorderRadius(tester), const BorderRadius.only(
+            topLeft: Radius.circular(4.0),
+            topRight: Radius.circular(4.0),
+          ));
+        });
 
-    testWidgets(
-      'The label appears above the input when there is no content and floatingLabelBehavior is always',
-      (WidgetTester tester) async {
-        await tester.pumpWidget(
-          buildInputDecorator(
-            isEmpty: true,
-            decoration: const InputDecoration(
-              labelText: labelText,
-              floatingLabelBehavior: FloatingLabelBehavior.always,
+        testWidgets('container has correct color', (WidgetTester tester) async {
+          await tester.pumpWidget(
+            buildInputDecorator(
+              decoration: const InputDecoration(
+                filled: true,
+                labelText: labelText,
+                helperText: helperText,
+              ),
             ),
-          ),
-        );
+          );
 
-        expect(getDecoratorRect(tester).size, const Size(800.0, 56.0));
-        expect(getLabelRect(tester).top, 8.0);
-        expect(getLabelRect(tester).bottom, 20.0);
-      },
-    );
+          final ThemeData theme = Theme.of(tester.element(findDecorator()));
+          expect(findBorderPainter(), paints
+            ..path(
+              style: PaintingStyle.fill,
+              color: theme.colorScheme.surfaceContainerHighest,
+            )
+          );
+        });
 
-    testWidgets(
-      'The label appears within the input text when there is content and floatingLabelBehavior is never',
-      (WidgetTester tester) async {
-        await tester.pumpWidget(
-          buildInputDecorator(
-            decoration: const InputDecoration(
-              labelText: labelText,
-              floatingLabelBehavior: FloatingLabelBehavior.never,
+        testWidgets('active indicator has correct height and color', (WidgetTester tester) async {
```

Comment:
Did you mean "weight" instead of "height"?

Reply:
Oh good catch!
For the active indicator 'height' would be acceptable because it is a simple horizontal line (the M3 specification uses 'height' for this particular value). But for outlined text field, 'height' is misleading when applied to the rectangular outline (funny enough, the M3 specification uses 'width' in this case).
Using "weigth" looks like the best choice, thanks!